jme3-core/src/main/java/com/jme3/light/SpotLight.java

```diff
@@ -58,19 +59,50 @@
  */
 public class SpotLight extends Light {
 
-    protected Vector3f position = new Vector3f();
-    protected Vector3f direction = new Vector3f(0,-1,0);
-    protected float spotInnerAngle = FastMath.QUARTER_PI / 8;
-    protected float spotOuterAngle = FastMath.QUARTER_PI / 6;
-    protected float spotRange = 100;
-    protected float invSpotRange = 1f / 100;
-    protected float packedAngleCos=0;
+    protected Vector3f position;
+    protected Vector3f direction;
+    protected float spotInnerAngle;
+    protected float spotOuterAngle;
+    protected float spotRange;
+    protected float invSpotRange;
+    protected float packedAngleCos;
     
     protected float outerAngleCosSqr, outerAngleSinSqr;
     protected float outerAngleSinRcp, outerAngleSin, outerAngleCos;
     
     public SpotLight() {
+        this(new ColorRGBA(1f, 1f, 1f, 1f), new Vector3f(), new Vector3f(0, -1, 0), FastMath.QUARTER_PI / 8, FastMath.QUARTER_PI / 6);
```

Comment:
Actually, if you submit your own color/direction, you are saving 2 object creations compared with the previous version. 
+The divisions should be resolved during compile time.